static/app/views/dashboards/widgetCard/autoSizedText.tsx

```diff
@@ -0,0 +1,145 @@
+      // The entries list contains an array of every observed item. Here it is only one element
+      const entry = entries[0];
+
+      if (!entry) {
+        return;
+      }
+
+      // The resize handler passes the parent's dimensions, so we don't have to get the bounding box
+      const parentDimensions = entry.contentRect;
+
+      // Reset the iteration parameters
+      fontSizeLowerBound.current = 0;
+      fontSizeUpperBound.current = parentDimensions.height;
+
+      let iterationCount = 0;
+
+      const span = Sentry.startInactiveSpan({
+        op: 'function',
+        name: 'AutoSizedText.iterate',
+      });
+
+      // Run the resize iteration in a loop. This blocks the main UI thread and prevents
+      // visible layout jitter. If this was done through a `ResizeObserver` or React State
+      // each step in the resize iteration would be visible to the user
+      while (iterationCount <= ITERATION_LIMIT) {
+        const childDimensions = getElementDimensions(childElement);
+
+        const widthDifference = parentDimensions.width - childDimensions.width;
+        const heightDifference = parentDimensions.height - childDimensions.height;
+
+        const childFitsIntoParent = heightDifference > 0 && widthDifference > 0;
+        const childIsWithinWidthTolerance =
+          Math.abs(widthDifference) <= maximumDifference;
+        const childIsWithinHeightTolerance =
+          Math.abs(heightDifference) <= maximumDifference;
+
+        if (
+          childFitsIntoParent &&
+          (childIsWithinWidthTolerance || childIsWithinHeightTolerance)
+        ) {
+          // Stop the iteration, we've found a fit!
+          span.setAttribute('widthDifference', widthDifference);
+          span.setAttribute('heightDifference', heightDifference);
+          break;
+        }
+
+        adjustFontSize(childDimensions, parentDimensions);
+
+        iterationCount += 1;
+      }
+
+      span.setAttribute('iterationCount', iterationCount);
+      span.end();
+    });
+
+    observer.observe(parentElement);
+
+    return () => {
+      observer.disconnect();
+    };
+  }, [maximumDifference]);
+
+  const adjustFontSize = (childDimensions: Dimensions, parentDimensions: Dimensions) => {
+    const childElement = childRef.current;
+
+    if (!childElement) {
+      return;
+    }
+
+    let newFontSize;
+
+    if (
+      childDimensions.width > parentDimensions.width ||
+      childDimensions.height > parentDimensions.height
+    ) {
+      // The element is bigger than the parent, scale down
+      newFontSize = (fontSizeLowerBound.current + fontSize.current) / 2;
+      fontSizeUpperBound.current = fontSize.current;
+    } else if (
+      childDimensions.width < parentDimensions.width ||
+      childDimensions.height < parentDimensions.height
+    ) {
+      // The element is smaller than the parent, scale up
+      newFontSize = (fontSizeUpperBound.current + fontSize.current) / 2;
```

Comment:
Thoughts on trimming or rounding the floating point values here? I dont think this is an issue tbh, but it might generate some really long floats here :D

Reply:
I could! Ostensibly modern fonts handle sub-pixel rendering gracefully enough. Are you worried about DOM bloat?

Reply:
I _think_ the browsers will handle it fine (though different platforms render text quite differently). I think its more likely just going to be an instance where someone inevitably sees a high precision float in the DOM and starts asking around or making up some wrong ideas, which is a conversation you can spare yourself from having :D